sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -166,14 +245,19 @@
                         errorMessage, exception, handler.getErrorContext(sender));
                     return Mono.error(error);
                 }
-                return send(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT);
-            });
+                return send(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT, transaction);
```

Comment:
Is there any reason to use `Mono<DeliveryState> send(..)` when all usages are `.then()`

Reply:
We need DeliveryState because we need to read the values from it and return to user.